core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java

```diff
@@ -243,7 +246,17 @@ public void configure() {
          if (rvrl != null) {
             rvrl.remoteValueNotFound(key);
          }
-         return null;
+         if (!hasSuccesfulResponse) {
+            RpcException rpcException = new RpcException("No successful response: " + responses);
```

Comment:
Actually it's a bit weird that remote node will return SuccessfulResponse{null}, I think that UnsuccessfulResponse would make more sense...